src/FHist.jl

```diff
@@ -16,34 +16,135 @@ using Base.Threads: SpinLock
 if !isdefined(Base, :get_extension)
     using Requires
 end
-#using Requires
 
 using BayesHistogram
 export BayesHistogram
 
 const _default_overflow = false
+_to_tuple(x::Tuple) = x
+_to_tuple(x) = tuple(x)
 
+_from_tuple(x::Tuple{Any}) = only(x)
+_from_tuple(x) = x
```

Comment:
we special case `Hist1D`, in the sense that the underlying data structure for `binedges` is still `NTuple{1, AbstractVector}`, but for all the user-facing APIs:
- `binedges`
- `bincenters`

we return just an `AbstractVector`, similarly, for constructor, we take just an `AbstractVector` as well